src/lib/Libnet/net_server.c

```diff
@@ -332,6 +332,23 @@ init_network_add(int sd, void (*readfunc)(int))
 		connection_init();
 		if (init_poll_context() < 0)
 			return (-1);
+#ifdef WIN32
+		/* set a dummy fd in the read set			*/
+		/* so that select() does not return WSAEVAL */
+		sd_dummy = socket(AF_INET, SOCK_STREAM, 0);
+		if (sd_dummy < 0) {
+			errno = WSAGetLastError();
+			log_err(errno, __func__, "socket() failed");
+			return (-1);
+		}
+		if((tpp_em_add_fd(poll_context, sd_dummy, EM_IN) == -1)){
+			int err = errno;
+			snprintf(logbuf, sizeof(logbuf), 
+				"Could not add socket %d to the read set", sd_dummy);
+			log_err(err, __func__, logbuf);
+			close_conn(sd);
+		}
+#endif /* WIN32 */
```

Comment:
This is fine actually, but it would be cleaner to add this code block to init_poll_context (under WIN32 of course), so that in case init_poll_context gets call elsewhere too the right thing happens. Thanks!

Reply:
Done.